src/json.rs

```diff
@@ -23,6 +27,9 @@ pub enum JsonError {
     InvalidString,
     InvalidArray,
     ExpectedStringGotNumber,
+    OutOfRangeU8,
+    OutOfRangeU64,
+    OutOfRangeU128,
```

Comment:
These should be condensed to be the same error `OutOfRange`.

Reply:
We should finalize Error details for aurora-engine. Someone wants more general errors, someone more detailed.
Actually, I think more detailed errors is a good way for an environment where impossible to debug.

Reply:
I mean, you could have another `Enum` of each kind if you want.

Reply:
I think it makes sense to follow conventions of other projects. And in my experience usually errors are grouped together as underflow and overflow (regardless of container type size).

That said, I agree more granular errors make debugging much easier when all we get usually is the error message. So I'm fine either way.

Reply:
But generally, these are all related to each other. So just add a field of which the max value is. That's pretty typical.

Reply:
@joshuajbouw I extended OutOfRange to separated kind of errors